hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -0,0 +1,590 @@
+                   to receive in this command.  This field is ignored when
+                   SPEED = 0, because the number of bytes received is always
+                   equal to the number of bytes transmitted in Standard SPI
+                   mode.'''
+          resval: "0x0"
+        },
+        { bits: "28:26",
+          name: "CSID",
+          desc: '''The Chip select ID of the device to receive the following
+                   command.''',
+          resval: "0x0",
+        },
+        { bits: "30:29",
+          name: "SPEED",
+          desc: '''The speed for the following command: "0" = Standard SPI. "1" = Dual SPI.
+                    "2"=Quad SPI,  "3": RESERVED.''',
+          resval: "0x0"
+        },
+        { bits: "31",
+          name: "GO",
+          hwqe: "true",
+          hwaccess: "hrw",
+          desc: '''Writing a one to this field finalizes the command and submits it to the SPI_HOST core.
+                   This field is self-clearing, so each write to this field only triggers one command.'''
+          resval: "0x0"
+        }
+      ]
+    },
+    { name: "TXDATA",
+      desc: "SPI Transmit data",
+      swaccess: "wo",
+      hwaccess: "hro",
+      hwqe: "true",
+      fields: [
+        { bits:  "31:24",
+          name:  "BYTE_D",
+          desc:  '''One of four bytes to transmit.  If ByteOrder = 0,
+                    this byte will be transmitted before BYTE_C, BYTE_B
+                    and BYTE_A. Otherwise BYTE_D will be transmitted
+                    last of the four.
+                    ''',
+          resval: "0x0"
+        },
+        { bits:  "23:16",
+          name:  "BYTE_C",
+          desc:  '''One of four bytes to transmit.  If ByteOrder = 0,
+                    this byte will be transmitted second, after BYTE_D.
+                    Otherwise BYTE_C will be transmitted third, after
+                    BYTE_A, and BYTE_B.
+                    ''',
+          resval: "0x0"
+        },
+        { bits:  "15:8",
+          name:  "BYTE_B",
+          desc:  '''One of four bytes to transmit.  If ByteOrder = 0,
+                    this byte will be transmitted third after BYTE_D and BYTE_C.
+                    Otherwise BYTE_B will be transmitted second, after BYTE_A.
+                    ''',
+          resval: "0x0"
+        },
+        { bits:  "7:0",
+          name:  "BYTE_A",
+          desc:  '''One of four bytes to transmit.  If ByteOrder = 0,
+                    This byte will be transmitted last, after BYTE_D, BYTE_C
+                    and BYTE_B.  Otherwise BYTE_A will be transmitted first.
+                    ''',
+          resval: "0x0"
+        }
+      ]
+    },
+    { name: "RXDATA",
+      desc: "SPI Receive data",
+      swaccess: "ro",
+      hwaccess: "hwo",
+      fields: [
+        { bits:  "31:24",
+          name:  "BYTE_D",
+          desc:  '''One of four bytes to receive.  If ByteOrder = 0,
```

Comment:
Assuming that SW doesn't need to predict the future, I think this should probably be "of four bytes received" :-D

Reply:
This one took me way too long to figure out what you were telling me.   Though in the end, I think I agree.